modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/SCIM2UserTestCase.java

```diff
@@ -325,6 +336,33 @@ private void validateFilteredUser(String attributeName, String operator, String
         assertEquals(userId, this.userId);
     }
 
+    private void validateFilteredUserByEmailAddresses(String attributeName, String operator, String attributeValue)
+            throws IOException {
+
+        String userResourcePath = getPath() + "?filter=" + USER_SYSTEM_SCHEMA_ATTRIBUTE + ":" + attributeName + operator
+                + attributeValue;
+        HttpGet request = new HttpGet(userResourcePath);
+        request.addHeader(HttpHeaders.AUTHORIZATION, getAuthzHeader());
+        request.addHeader(HttpHeaders.CONTENT_TYPE, "application/json");
+        request.addHeader(HttpHeaders.ACCEPT, "application/scim+json");
+        HttpResponse response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 200, "User " +
+                "has not been retrieved successfully");
+
+        Object responseObj = JSONValue.parse(EntityUtils.toString(response.getEntity()));
+        EntityUtils.consume(response.getEntity());
+
+        JSONArray emailsArray = ((JSONArray)((JSONObject)((JSONObject) ((JSONArray) ((JSONObject) responseObj)
+                .get("Resources")).get(0)).get(USER_SYSTEM_SCHEMA_ATTRIBUTE)).get(attributeName));
+
+        for (Object email : emailsArray) {
```

Comment:
```suggestion
        for (Object email: emailsArray) {
```

Reply:
Fixed by - https://github.com/wso2/product-is/pull/23619/commits/c0196674373dfa6ad281ac00dca19336d752abdb